common_module.py

```diff
@@ -248,9 +248,21 @@ def __init__(self, fn):
         self.fn = fn
 
 
+class FileEmptyError(Exception):
+    def __init__(self, fn):
+        msg = "File %s is empty.\n" \
+              "Is your batch scheduler loaded with jobs?" % fn
+        Exception.__init__(self, msg)
+        logging.warning(msg)
+        self.fn = fn
```

Comment:
I find it difficult to believe there won't be code repetition with other similar exception handlers, but ok for now.

Overall, the PR looks good to me, although I didn't dive into its details fully... merging anyhow